### DISCUSSION TITLE
[@types/react]  React.Key `bigint` issues

### DISCUSSION THREAD

**Discussion by bombillazo** [DefinitelyTyped/DefinitelyTyped#66749] [Issues with a @types package]
The addition of `bigint` to the `React.Key` type now complicates the usage of Key in different scenarios. For example, we are using a field of React.Key type to index an object, but can no longer do this since bigint is not a valid index:
```ts
export type MyType = {
  [idx: string| number]: number;
};

export type MyType = {
  [idx: React.Key]: number;
};
```

![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/44002321/7fddcd14-f5d8-464f-aff4-ea069582debd)

Another issue is when trying to pass a React.Key value to an Input component that uses the  `InputHTMLAttributes` input type:
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/44002321/c50826ec-86bb-430d-ab47-fcf4afe5de7d)
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/44002321/c12891de-0211-4fce-968a-53a341e6b001)

[a comment by an automated account is not shown]

**Comment by ilteoood** [comment]
I'm experiencing the same issue: 
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/6383527/b0da7280-a7b4-4865-9edb-6c21aa75f54e)

**Reply by pvdstel** [reply]
I ran into a similar issue, but in my case it was caused by dependencies also including `@types/react`. With this change, they were no longer compatible. So try and see if your nested `@types/react` dependencies are up to date :smile: that fixed it here.

**Comment by pvdstel** [comment]
`React.Key` is probably not meant to be used for indexing objects, just for identifying elements uniquely. It might be better to define your own key type there. 🙂

**Comment by eps1lon** [comment]
>  For example, we are using a field of React.Key type to index an object, but can no longer do this since bigint is not a valid index:

It was not intended for that. You should use `string | number | null` if you want the old structure.

`React.Key` supposed to signal what the `key` prop accepts. If you use any type alias for its structure, not it's meaning, you're going to encounter a lot more issues going forward.
